contracts/BaalSummoner.sol

```diff
@@ -4,47 +4,63 @@ pragma solidity 0.8.7;
 import "@gnosis.pm/zodiac/contracts/factory/ModuleProxyFactory.sol";
 import "@gnosis.pm/safe-contracts/contracts/proxies/GnosisSafeProxyFactory.sol";
 import "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 
 import "./Baal.sol";
 
-contract BaalSummoner is ModuleProxyFactory {
-    address payable public immutable template; // fixed template for baal using eip-1167 proxy pattern
+contract BaalSummoner is Initializable, OwnableUpgradeable, UUPSUpgradeable {
+    // when some of the init addresses are updated
+    uint256 public addrsVersion;
+
+    address payable public template; // fixed template for baal using eip-1167 proxy pattern
 
     // Template contract to use for new Gnosis safe proxies
-    address public immutable gnosisSingleton;
+    address public gnosisSingleton;
 
     // Library to use for EIP1271 compatability
-    address public immutable gnosisFallbackLibrary;
+    address public gnosisFallbackLibrary;
 
     // Library to use for all safe transaction executions
-    address public immutable gnosisMultisendLibrary;
+    address public gnosisMultisendLibrary;
 
     // template contract to clone for loot ERC20 token
-    address public immutable lootSingleton;
+    address public lootSingleton;
 
     // template contract to clone for shares ERC20 token
-    address public immutable sharesSingleton;
```

Comment:
wondering what the reason is for removing immutable?

Reply:
this contract is now upgradable and there is a new owner only function to change these addresses.